scripts/test_sql_migration_downgrade.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+set -Eeuo pipefail
+
+BOLDGREEN="\033[1;32m"
+ENDCOLOR="\033[0m"
+
+export FLASK_APP=application.py
+export NOTIFY_ENVIRONMENT="development"
+
+# what is the database currently pointing at on main?
+CURRENT_VERSION=$(curl -s https://raw.githubusercontent.com/alphagov/notifications-api/main/migrations/.current-alembic-head)
+
+# what is the most recent version? note this depends on us only ever having one head
+NEWEST_VERSION=$(flask db heads | tail -n 1 | cut -d " " -f 1)
+
+if [[ "${CURRENT_VERSION}" == "${NEWEST_VERSION}" ]]; then
+    echo "No migrations to check"
+    exit 0;
+fi
+
+
+# delete any existing test DB
+psql -c "drop database migration_test" || true
+psql -c "create database migration_test"
+
+echo -e "${BOLDGREEN}======== running upgrade (logs hidden) ========${ENDCOLOR}"
+
+SQLALCHEMY_DATABASE_URI="postgresql://localhost/migration_test" flask db upgrade > /dev/null 2>&1
+
+# make sure downgrade can run succesfully
+echo -e "${BOLDGREEN}============== running downgrade ==============${ENDCOLOR}"
+SQLALCHEMY_DATABASE_URI="postgresql://localhost/migration_test" flask db downgrade
+
+# make sure downgrade has left everything okay
+echo -e "${BOLDGREEN}============ running upgrade again ============${ENDCOLOR}"
+SQLALCHEMY_DATABASE_URI="postgresql://localhost/migration_test" flask db upgrade $CURRENT_VERSION
```

Comment:
It's not clear to me why we're doing `flask db upgrade $CURRENT_VERSION` here and not `flask db upgrade`? With this line as it is, we're not re-running the migration that was added, it's just trying to make the db match the version on main, so I'm seeing this script pass with a migration file I would expect to fail

Reply:
oops that's a typo - i meant to add this line to `flask db downgrade` (so that if theres more than one downgrade script in a PR, it'll test all of them)

Reply:
fixed in fe6f9bf14